internal/settings/providers/servercow/provider.go

```diff
@@ -0,0 +1,179 @@
+package servercow
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/qdm12/ddns-updater/internal/models"
+	"github.com/qdm12/ddns-updater/internal/settings/constants"
+	"github.com/qdm12/ddns-updater/internal/settings/errors"
+	"github.com/qdm12/ddns-updater/internal/settings/headers"
+	"github.com/qdm12/ddns-updater/internal/settings/log"
+	"github.com/qdm12/ddns-updater/internal/settings/utils"
+	"github.com/qdm12/ddns-updater/pkg/publicip/ipversion"
+)
+
+type provider struct {
+	username      string
+	host          string
+	domain        string
+	ipVersion     ipversion.IPVersion
+	password      string
+	useProviderIP bool
+	ttl           uint
+	logger        log.Logger
+}
+
+func New(data json.RawMessage, domain, host string, ipVersion ipversion.IPVersion, logger log.Logger) (p *provider, err error) {
```

Comment:
The CI is failing due to linting as the 'line is too long', you can just commit that suggestion and it *should* pass

```suggestion
func New(data json.RawMessage, domain, host string, ipVersion ipversion.IPVersion,
	logger log.Logger) (p *provider, err error) {
```